spotify_tensorflow/dataframe_endpoint.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import logging
+import sys
+import timeit
+
+import pandas as pd
+import six
+import tensorflow as tf
+
+from .dataset import Datasets
+
+FLAGS = tf.flags.FLAGS
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DataFrameEndpoint(object):
+    @classmethod
+    def read_dataset(cls, dataset_path,
+                     take=sys.maxsize,
+                     unpack_multispec=False,
+                     as_np=False,
```

Comment:
Does it make sense to have this as_np argument?  End users can call as_matrix externally also and deal with issues there.

Reply:
that's a fair point. I wanted to have it in the API because most models consume numpy arrays directly. 
I'm actually wondering if it makes sense to expose the Pandas DF at all (as opposed to numpy array). The main reasons I picked pandas is because it's easy to parse TF dataset and also the users see the feature names. 
What do you think about exposing only numpy arrays? and possibly the feature names as a second object.

Reply:
I think pandas may be helpful.  Also more generalizable.  Could be helpful if people want to remove some rows etc when debugging.  Especially if you want to run sum summary stats on the dataframe.

Reply:
alright, I'll rm the np helper and perhaps add it back if 100% of users end up converting to numpy